src/containers/settings/EditServiceScreen.tsx

```diff
@@ -188,88 +189,117 @@ class EditServiceScreen extends Component<IProps> {
         },
         isEnabled: {
           label: intl.formatMessage(messages.enableService),
-          value: service?.isEnabled,
+          value: ifUndefined<boolean>(
+            service?.isEnabled,
+            DEFAULT_SERVICE_SETTINGS.isEnabled,
+          ),
           default: DEFAULT_SERVICE_SETTINGS.isEnabled,
           type: 'checkbox',
         },
         isHibernationEnabled: {
           label: intl.formatMessage(messages.enableHibernation),
-          value: service?.isHibernationEnabled,
+          value: ifUndefined<boolean>(
+            service?.isHibernationEnabled,
+            DEFAULT_SERVICE_SETTINGS.isHibernationEnabled,
+          ),
           default: DEFAULT_SERVICE_SETTINGS.isHibernationEnabled,
           type: 'checkbox',
         },
         isWakeUpEnabled: {
           label: intl.formatMessage(messages.enableWakeUp),
-          value: service?.isWakeUpEnabled,
+          value: ifUndefined<boolean>(
+            service?.isWakeUpEnabled,
+            DEFAULT_SERVICE_SETTINGS.isWakeUpEnabled,
+          ),
           default: DEFAULT_SERVICE_SETTINGS.isWakeUpEnabled,
           type: 'checkbox',
         },
         isNotificationEnabled: {
           label: intl.formatMessage(messages.enableNotification),
-          value: service?.isNotificationEnabled,
+          value: ifUndefined<boolean>(
+            service?.isNotificationEnabled,
+            DEFAULT_SERVICE_SETTINGS.isNotificationEnabled,
+          ),
           default: DEFAULT_SERVICE_SETTINGS.isNotificationEnabled,
           type: 'checkbox',
         },
         isBadgeEnabled: {
           label: intl.formatMessage(messages.enableBadge),
-          value: service?.isBadgeEnabled,
+          value: ifUndefined<boolean>(
+            service?.isBadgeEnabled,
+            DEFAULT_SERVICE_SETTINGS.isBadgeEnabled,
+          ),
           default: DEFAULT_SERVICE_SETTINGS.isBadgeEnabled,
           type: 'checkbox',
         },
         isMediaBadgeEnabled: {
           label: intl.formatMessage(messages.enableMediaBadge),
-          value: service?.isMediaBadgeEnabled,
+          value: ifUndefined<boolean>(
+            service?.isMediaBadgeEnabled,
+            DEFAULT_SERVICE_SETTINGS.isMediaBadgeEnabled,
+          ),
           default: DEFAULT_SERVICE_SETTINGS.isMediaBadgeEnabled,
           type: 'checkbox',
         },
         trapLinkClicks: {
           label: intl.formatMessage(messages.trapLinkClicks),
-          value: service?.trapLinkClicks,
+          value: ifUndefined<boolean>(
+            service?.trapLinkClicks,
+            DEFAULT_SERVICE_SETTINGS.trapLinkClicks,
+          ),
           default: DEFAULT_SERVICE_SETTINGS.trapLinkClicks,
           type: 'checkbox',
         },
         isMuted: {
           label: intl.formatMessage(messages.enableAudio),
-          value: !service?.isMuted,
+          value: ifUndefined<boolean>(
```

Comment:
This gives the opposite value as expected since the checkbox is about enabling the audio, but the boolean is about being muted. So this should be `! ifUndefined...` in the same way the negation was there before.

Reply:
thanks for catching that - fixed in the latest commit